Sources/SKCore/FileBuildSettingsChange.swift

```diff
@@ -16,8 +16,16 @@ public enum FileBuildSettingsChange {
   /// The `BuildSystem` has no `FileBuildSettings` for the file.
   case removedOrUnavailable
 
-  /// The `FileBuildSettings` have been modified.
+  /// The `FileBuildSettings` have been modified or are newly available.
   case modified(FileBuildSettings)
+
+  /// The `BuildSystem` is providing fallback arguments which may not be correct,
+  /// or potentially is missing some generated inputs (e.g. prebuilding required).
```

Comment:
> or potentially is missing some generated inputs (e.g. prebuilding required).

This is a state that is independent of the settings themselves, and I would not expect it to be represented as `fallback`.  It feels like this is trying to shoehorn a more general status into the settings.

Reply:
Yeah I thought it made sense to combine them since a file status containing `generated inputs not available` would mean a very similar thing to the fallback state to the LSP instances themselves. If we separate it out into the status, that means that the status the build system reports will also impact the ASTs and force reloads.

With this approach here, the individual LSPs manage the file state, and we could maybe have the build system just report a textual status update, no AST impact. I'm trying to keep the future file status implementation in mind.

Reply:
If you are missing dependencies, you may *want* to see that reflected in your diagnostics as e.g. missing modules. If your editor doesn't support the file status extension, this may be the only indication of what has gone wrong, since otherwise you just silently lose semantic functionality.  If you later build and then start seeing errors that weren't there before, I think that may be confusing.  This is especially painful if you have  dependencies built and then they are removed (i.e. build is cleaned), so the state goes from showing diagnostics to hiding them.  Without the file status extension, that's very confusing.

For fallback settings, you are in a more or less permanent state (so it won't confusing jump between errors and no errors).